### PR TITLE
Minisensor 2023 cleanup

### PR DESCRIPTION
**Pull request by BluCodeGH**
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/cansw_sensor/6)

### CODE REVIEW

**Review by BluCodeGH — COMMENTED** [review]
> Reviewed 3 of 10 files at r1, 14 of 14 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/cansw_sensor/6)* status: all files reviewed, 1 unresolved discussion (waiting on @Darth-Raazi and @rdakhil)

___
*[`main.c` line 132 at r4](https://reviewable.io/reviews/waterloo-rocketry/cansw_sensor/6#-NVghwXg696p8DemtSan:-NVghwXg696p8DemtSao:bepvpzl) ([raw file](https://github.com/waterloo-rocketry/cansw_sensor/blob/11587d6c3329c457ffa9799774b75b15761861ca/main.c#L132)):*
> ```C
> 
>             can_msg_t sensor_msg;
>             build_analog_data_msg(millis(), SENSOR_PRESSURE_CC, pressure_4_20_psi, &sensor_msg);
> ```

Can we find some way to not hardcode the sensor IDs and which sensors are connected? Also pneumatics and CC/ox tank should probably have different configurable rates rather than one global PRES_TIME.

File: ICM-20948.c [hunk 1]
```diff
@@ -0,0 +1,141 @@
+#ifndef ICM_20948_H
```
Comment (JasonBrave) [inline ICM-20948.c]: Header guards should not exist in C source files

File: main.c [hunk 2]
```diff
@@ -17,20 +17,25 @@
...
-#define BARO_TIME_DIFF_ms 500
-#define IMU_TIME_DIFF_ms 0
-#define PRES_TIME_DIFF_ms 500
-#define TEMP_TIME_DIFF_ms 0
+#define STATUS_TIME_DIFF_ms 500 // 2 Hz
+#define BARO_TIME_DIFF_ms 500 // 2 Hz
+#define IMU_TIME_DIFF_ms 500 // 2 Hz
+#define PRES_OX_CC_TIME_DIFF_ms 500 // 2 Hz
+#define PRES_PNEUMATICS_TIME_DIFF_ms 500 // 2 Hz
+#define TEMP_TIME_DIFF_ms 0 // 2 Hz
```
Comment (JasonBrave) [inline main.c]: Comment should reflect macro setting in the code, for this case it should be "Disabled"

**Review by JasonBrave — CHANGES_REQUESTED** [review]

File: ICM-20948.c [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    uint8_t y_h = MY2C_read1ByteRegister(AK09916_mag_addr, HYH);
+    uint8_t y_l = MY2C_read1ByteRegister(AK09916_mag_addr, HYL);
+    *y = (int16_t)((uint16_t)y_h << 8 | y_l);
+
+    uint8_t z_h = MY2C_read1ByteRegister(AK09916_mag_addr, HZH);
+    uint8_t z_l = MY2C_read1ByteRegister(AK09916_mag_addr, HZL);
+    *z = (int16_t)((uint16_t)z_h << 8 | z_l);
+
+    // Must read ST2 register after measurement, see datasheet register 13.4 ST2
+    uint8_t mag_status_2 = MY2C_read1ByteRegister(AK09916_mag_addr, ST2);
```
Comment (JasonBrave) [inline ICM-20948.c]: Unused variable

**Review by JasonBrave — CHANGES_REQUESTED** [review]

**Comment by JasonBrave** [comment]
Lint warnings fixed

**Review by JasonBrave — APPROVED** [review]
LGTM

**Pull request merged**